### DISCUSSION TITLE
Create file content API is not work for hidden directory

### DISCUSSION THREAD

**Discussion by radhikari-arch** [community/community#41769] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

Hello,
I tried a create content using API from here:
https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents
I am unable to create the files that contains .github/workflow. Here is an API url:
```
https://api.github.com/repos/{{ organization_name }}/{{github_repo_name}}/contents/.github/workflows/continuous_integration.yml
```

Error received:
```"Resource not accessible by integration"}, "msg": "Status code was 403 and not [201]: HTTP Error 403: Forbidden```

**Comment by radhikari-arch** [comment]
If someone is testing this, we used the github app, and I believe has the correct permission as I am able to create other files, even the hidden files. It is just the in the .github/workflow directory.
```
"Resource not accessible by integration"}, "msg": "Status code was 403 and not [201]: HTTP Error 403: Forbidden
```

**Reply by OCTOBRIGHT** [reply]
Thank you I don't understand any of this so it's going to be a minute for me because I don't even know how to do code I may have to get a book or something my first respiratory someone took over and I never even knew that anything about this and now I do I'm starting to learn thank you

**Comment by airtower-luna** [comment, marked as answer]
There's a separate permission for editing workflow files, which is needed in addition to general write access to the repository. For [OAuth apps](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes) the scope is called *workflow*, I'm not sure what the GitHub app equivalent is called. Make sure your application [has that permission](https://docs.github.com/en/developers/apps/managing-github-apps/editing-a-github-apps-permissions).

**Reply by radhikari-arch** [reply]
Could not figure out the permissions needed. I have at least these read-write permission which seems to be relevant permissions:
- action (Workflows, workflow runs and artifacts.)
- Contents (Repository contents, commits, branches, downloads, releases, and merges.)
- Workflows (Update GitHub Action workflow files.)

Also, testing again, I am able to upload any other hidden directory, just not the .github/workflow.

**Reply by radhikari-arch** [reply]
NM, this worked. Just saving the apps setting was not enough, had to accept the changes as well. Workflows (Update GitHub Action workflow files.) was the required permission.
